src/app/api/users/route.ts

```diff
@@ -0,0 +1,30 @@
+import { user } from "@/dao/user.dao";
+import { NextResponse, NextRequest } from "next/server";
+import { z } from "zod";
+
+const checkGetAllUsersParams = z.object({
```

Comment:
Let's give this schema a better name: 

```suggestion
const ListAllUsersParamsSchema = z.object({
```

Zod is based on `schema` definition. We're in fact using it for `check`ing the data we're getting, but we should always name zod objects as `schema`